app/_src/gateway-operator/production/upgrade/data-plane/blue-green.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Blue/Green Upgrades
+---
+
+## Using DataPlane
+
+Blue/Green upgrades can be accomplished when working with the `DataPlane` resource directly.
+
+1. To enable blue/green deployments set the `spec.deployment.rollout.strategy` on your `DataPlane` resource:
+
+    ```yaml
+    apiVersion: gateway-operator.konghq.com/v1beta1
+    kind: DataPlane
+    metadata:
+      name: bluegreen
+    spec:
+      deployment:
+        rollout:
+          strategy:
+            blueGreen:
+              promotion:
+                strategy: BreakBeforePromotion
+        podTemplateSpec:
+          spec:
+            containers:
+            - name: proxy
+              image: kong/kong-gateway:{{ site.data.kong_latest_gateway.ee-version }}
+              env:
+              - name: KONG_LOG_LEVEL
+                value: debug
+              readinessProbe:
+                initialDelaySeconds: 1
+                periodSeconds: 1
+    ```
+
+    > **NOTE**: Currently only `BreakBeforePromotion` is available as promotion strategy.
+
+    When applied like this, {{ site.kgo_product_name }} will deploy new `Services` through which you'll be able to access new `Pod`s once they are available.
+
+    By default no `Pod`s will be deployed immediately, instead {{ site.kgo_product_name }} will observe the `DataPlane` resource for `spec` changes and when any configuration drift is detected it will spawn a new "preview" `Deployment` which will contain the changes applied to the `DataPlane` resource.
+
+1. Wait for `DataPlane` to be ready to accept changes
+
+    ```bash
+    kubectl wait dataplane bluegreen --for=jsonpath='{.status.rollout.conditions[*].reason}'=WaitingForChange
+    ```
+
+1. Test it out by patching the `DataPlane` with a new `image`:
+
+    ```bash
+    $ kubectl patch dataplane bluegreen --type='json' -p='[{"op": "replace", "path": "/spec/deployment/podTemplateSpec/spec/containers/0/image", "value":"kong:3.3.1"}]'
+    dataplane.gateway-operator.konghq.com/bluegreen patched
+    ```
+
+    After this patch gets applied you'll be able to access the new {{ site.base_gateway }} `Pod`s via the "preview" ingress `Service`.
+
+1. To find the "preview" `Service` you can look up `DataPlane` status, and more specifically its `rollout` field:
+
+    ```bash
+    $ kubectl get dataplane bluegreen -o jsonpath-as-json='{.status.rollout}
```

Comment:
The results should look like this:
```
output here
```

Reply:
This has been split into 2 blocks: command and output.